crates/router/src/core/admin.rs

```diff
@@ -466,13 +466,17 @@ pub async fn merchant_account_delete(
     merchant_id: String,
 ) -> RouterResponse<api::MerchantAccountDeleteResponse> {
     let db = state.store.as_ref();
-    let is_deleted = db
+    let is_merchant_account_deleted = db
         .delete_merchant_account_by_merchant_id(&merchant_id)
         .await
         .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+    let is_merchant_key_store_deleted = db
+        .delete_merchant_key_store_by_merchant_id(&merchant_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
```

Comment:
Better to perform this operation only if the merchant account was deleted, you can have a condition here as to 
```rust
if is_merchant_account_deleted {
    // Delete merchant_key_store
}